src/docs/sphinx/dev_guide/release.rst

```diff
@@ -103,72 +102,105 @@ Typical changes that are made in a release candidate branch include:
+will be much easier since it will show only the changes made in the release
+candidate that are not in the develop branch.
+
+To facilitate the review process, we cross reference the two pull requests
+in their respective pull request descriptions. Suppose the pull request for
+the release (to merge the release candidate into main) is #N and the
+companion pull request (to merge the release candidate into develop) is #M.
+In the description of pull request N, add a link to pull request M and a
+comment to **review pull request M and approve pull request N.** In the
+description of pull request M, add a link to pull request M and a statement
+that it is the companion pull request for N, that it **should be reviewed and
+not merged** and that pull request N should be approved as it will be merged
+for the release.
+
+4: Merge Release Candidate
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Merge the release candidate branch into the main branch once it is ready and
-approved. Do not "squash merge" as it will make the histories of main and
-develop branches disagree, and we want to preserve the history. After
-merging, the release candidate branch can be deleted.
+Merge the release candidate branch into the main branch after it has been
+approved and all CI checks have passed. Do not "squash merge" as it will make
+the histories of main and develop branches disagree, and we want to preserve
+the history. After merging, the release candidate branch can be deleted.
 
-
-4: Draft a GitHub Release
+5: Draft a GitHub Release
 ^^^^^^^^^^^^^^^^^^^^^^^^^
 
 `Draft a new Release on GitHub <https://github.com/LLNL/axom/releases/new>`_
 
-#. Enter the desired tag version, e.g., v0.5.0
+#. Enter a Release title. We typically use titles of the following form *Axom-v0.3.1*
 
 #. Select **main** as the target branch to tag a release.
 
-#. Enter a Release title. We typically use titles of the following form *Axom-v0.3.1*
+#. Enter the release tag name, such as v0.5.0, and specify to create the tag 
+   when the release is published.
 
 #. Copy and paste the information for the release from the
-   ``axom/RELEASE-NOTES.md`` into the release description (omit any sections if empty).
+   ``axom/RELEASE-NOTES.md`` into the release description 
+   (omit any sections that are empty).
+
+#. Add a statement at the top of the release note, such as:
+   Please download the `Axom-*.tar.gz` tarball below, which includes all of the
+   submodules needed to build Axom. The `AxomData-*.tar.gz` tarball contains
+   extra files to put in the `data` directory only if you want to run certain
+   Axom tests.
 
 #. Publish the release. This will create a tag at the tip of the main
    branch and add corresponding entry in the
    `Releases section <https://github.com/LLNL/axom/releases>`_
 
 .. note::
 
-   GitHub will add a corresponding tarball and zip archives consisting of the
+   GitHub will add tarfile and zip archives consisting of the
```

Comment:
Is "tarfile" now preferred over "tarball?"

Reply:
Not particularly. I prefer tarfile. "tarball" reminds me of an old software engineering paper titled "Big Ball of Mud" https://www.researchgate.net/publication/2938621_Big_Ball_of_Mud